dromedary.py

```diff
@@ -163,6 +180,166 @@ def print_output(out: subprocess.CompletedProcess) -> None:
     print(std_out + err_out)
 
 
+###############################################################################
+def run_cmd_output(cmd_args: List[Any], cmd_env: Optional[Dict[str, str]]) -> str:
+    """Run the given command in a shell.
+
+    Prints the output of the process in "real time".
+
+    Args:
+        cmd_args (List[Any]): The command and it's arguments to run.
+        cmd_env (Optional[Dict[str, str]]): The environment to run the command
+                                            in.
+
+    Returns:
+        str: The accumulated stdout and stderr output of the process.
+    """
+    std_out: str = ""
+    with subprocess.Popen(
+        " ".join(map(lambda e: f"'{str(e)}'", cmd_args)),
+        shell=True,
+        env=cmd_env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    ) as p:  # nosec
+        while p.poll() is None:
+            std_str = p.stdout.read1().decode("utf-8")
+            std_out += std_str
+            print(std_str, end="", flush=True, file=sys.stdout)
+
+        if p.poll() != 0:
+            print(f"Error: command '{p.args}' returned '{p.poll()}'")
+            return error_exit(6)
+
+    return std_out
+
+
+class SwitchConfig(NamedTuple):
+    """Holds valid configuration of a switch to create."""
+    name: str
+    compiler: str
+    packages: List[str]
+
+
+###############################################################################
+def generate_switch(config_file: str) -> Tuple[str, List[str]]:
+    """Generate an Opam switch from the configuration in the given JSON file
+    `config_file`.
+
+    Args:
+        config_file (str): The path to the JSON config file to use.
+
+    Returns:
+        SwitchConfig: A `Tuple` containing the name of the generated Opam
+                      switch and the list of packages to install.
+    """
+    print(f"Using config file '{config_file}'")
```

Comment:
Feedback from @grievejia 

I believe the usual convention is that human-oriented helper messages like this go to stderr, whereas stdout is strictly reserved for the the "proper" output of the script? Same for all other occurrences of print() in this diff.

Reply:
The idea behind that convention is to keep messages for the user separate from the output of the program that is, for example, piped into another program. But `dromedary.py` does not produce data that is meant to be parsed or piped into another program at all, that's what its generated output file and the optionally created intermediate JSON file are for. So, instead of making the script less usable for a human user by printing non-errors to stderr, I rather add an option to output machine-readable data.